packages/esbuild/src/build-cache.ts

```diff
@@ -0,0 +1,55 @@
+import { statSync } from 'fs';
+import type { OnLoadResult } from 'esbuild';
+
+// mtimes of known files shared between builds
+const fileInfo = new Map<string, number>();
+
+export function buildCache() {
+    const fileInfoDuringBuild = new Map<string, number>();
+    const cache = new Map<string, OnLoadResult>();
+    const checkCache = (path: string) => {
+        const cached = cache.get(path);
+        if (cached) {
+            const shouldRebuild = cached.watchFiles?.some((path) => {
+                const num = fileInfo.get(path);
+                if (num) {
+                    let stat = fileInfoDuringBuild.get(path);
+                    if (stat === undefined) {
+                        try {
+                            stat = statSync(path).mtimeMs;
+                            fileInfoDuringBuild.set(path, stat);
+                        } catch (e) {
+                            fileInfo.delete(path);
+                            return true;
+                        }
+                    }
+                    return stat !== num;
+                } else {
+                    return true;
+                }
+            });
+            if (!shouldRebuild) {
+                return cached;
+            }
+        }
+        return undefined;
+    };
+    const addToCache = (path: string, result: OnLoadResult) => {
+        cache.set(path, result);
+        for (const watchFile of result.watchFiles || []) {
+            if (fileInfo.has(watchFile)) {
+                continue;
+            }
+            fileInfo.set(watchFile, statSync(watchFile).mtimeMs);
```

Comment:
did you check that stylable breaks the build in case a dependency doesn't exist in a way that won't reach this point?